ROX-18614: add multi-tenancy guide (#78917)
openshift/openshift-docs: operating/manage-user-access/understanding-multi-tenancy.adoc

The following example combines a custom permission set, access scope, and role. The user or group assigned with this role can only see CVE information, violations, and information about deployments in the particular namespace or cluster scoped to them.

.Procedure

. In the {product-title-short} portal, select *Platform Configuration* -> *Access Control*.
. Select *Permission Sets*.
. Click *Create permission set*.
. Enter a *Name* and *Description* for the permission set.
. Select the following resources and access level and click *Save*:
** `READ` Alert
** `READ` Deployment
** `READ` DeploymentExtension
** `READ` Image
** `READ` K8sRole
** `READ` K8sRoleBinding
** `READ` K8sSubject
** `READ` NetworkGraph
** `READ` NetworkPolicy
** `READ` Secret
** `READ` ServiceAccount
. Select *Access Scopes*.
. Click *Create access scope*.
. Enter a *Name* and *Description* for the access scope.
. In the *Allowed resources* section, select the namespace you want to use for scoping and click *Save*.
. Select *Roles*.
. Click *Create role*.
. Enter a *Name* and *Description* for the role.
. Select the previously created *Permission Set* and *Access scope* for the role and click *Save*.

> . Assign the role to your required user or group. See xref:../../operating/manage-user-access/manage-role-based-access-control-3630.adoc#assign-role-to-user-or-group_manage-role-based-access-control[Assign a role to a user or a group].

Reviewer: ```suggestion
. Assign the role to your required user or group. See xref:../../operating/manage-user-access/manage-role-based-access-control-3630.adoc#assign-role-to-user-or-group_manage-role-based-access-control[Assigning a role to a user or a group].
```

:_mod-docs-content-type: ASSEMBLY
[id="understanding-multi-tenancy"]
= Understanding multi-tenancy
include::modules/common-attributes.adoc[]
:context: understanding-multi-tenancy

toc::[]

[role="_abstract"]

{product-title-short} provides ways to realize multi-tenancy within a Central instance.

This is being done by utilizing role-based access control (RBAC) and access scopes within {product-title-short}.

> [role="_additional-resources"]

Reviewer: I would move the additional resources section to the end unless there is a specific reason that it is here at the beginning.

* xref:../../operating/manage-user-access/manage-role-based-access-control-3630.adoc#create-a-custom-role-3630_manage-role-based-access-control[Create a custom role]

[id="understanding-resource-scoping"]
== Understanding resource scoping

{product-title-short} includes resources which are used within RBAC.
In addition to associating permissions for a resource, each resource is also scoped.

In {product-title-short}, resources are scoped as:

* Global scope, where a resource is not assigned to any cluster or namespace.
* Cluster scope, where a resource is assigned to particular cluster(s).
* Namespace scope, where a resource is assigned to particular namespace(s).

The scope of resources is important when creating custom access scopes.
Custom access scopes are used to create multi-tenancy within {product-title-short}.

Only resources which are cluster or namespace scoped are applicable for scoping in access scopes. Globally scoped resources are not scoped by access scopes.

Thus, multi-tenancy within {product-title-short} can only be achieved for resources that are scoped either by cluster or namespace.

[id="multi-tenancy-for-namespace"]

> == Multi-tenancy per namespace

Reviewer: ```suggestion
== Multi-tenancy per namespace configuration example
```

> One of the most common examples for multi-tenancy within {product-title-short} is associating users with a specific namespaces and only allowing them access to their specific namespace.

Reviewer: ```suggestion
A common example for multi-tenancy within {product-title-short} is associating users with a specific namespace and only allowing them access to their specific namespace.
```

> Below is an example of combining a custom permission set, access scope, and role. The user or group assigned with this role will only be able to see CVE information, violations, and information about deployments in the particular namespace or cluster scoped to them.

Reviewer: ```suggestion
The following example combines a custom permission set, access scope, and role. The user or group assigned with this role can only see CVE information, violations, and information about deployments in the particular namespace or cluster scoped to them.
```

.Procedure

. In the {product-title-short} portal, select *Platform Configuration* -> *Access Control*.
. Select *Permission Sets*.
. Click *Create permission set*.
. Enter a *Name* and *Description* for the permission set.
. Select the following resources and access level and click *Save*:
** `READ` Alert
** `READ` Deployment
** `READ` DeploymentExtension
** `READ` Image
** `READ` K8sRole
** `READ` K8sRoleBinding
** `READ` K8sSubject
** `READ` NetworkGraph
** `READ` NetworkPolicy
** `READ` Secret
** `READ` ServiceAccount
. Select *Access Scopes*.
. Click *Create access scope*.
. Enter a *Name* and *Description* for the access scope.

> . In the *Allowed resources* select the namespace you want to use for scoping and click *Save*.

Reviewer: ```suggestion
. In the *Allowed resources* section, select the namespace you want to use for scoping and click *Save*.
```

. Select *Roles*.
. Click *Create role*.
. Enter a *Name* and *Description* for the role.
. Select the previously created *Permission Set* and *Access scope* for the role and click *Save*.

> . Assign the role to your desired user or group following xref:../../operating/manage-user-access/manage-role-based-access-control-3630.adoc#assign-role-to-user-or-group_manage-role-based-access-control[these instructions].

Reviewer: ```suggestion
. Assign the role to your required user or group. See xref:../../operating/manage-user-access/manage-role-based-access-control-3630.adoc#assign-role-to-user-or-group_manage-role-based-access-control[Assigning a role to a user or a group].
```
[IBM Style](https://www.ibm.com/docs/en/ibm-style?topic=word-usage) disallows use of "desired"
Also we have to use the title of the section or doc in the links

[NOTE]
====

> The {product-title-short} dashboard for users with the sample role will be minimal compared to that of an administrator.

Reviewer: ```suggestion
The {product-title-short} dashboard options for users with the sample role are minimal compared to options available to an administrator.
```
IBM Style - use present tense where possible, not future

> Only relevant pages will be visible for the user.

Reviewer: ```suggestion
Only relevant pages are visible for the user.
```
keep this sentence with previous sentence since this is a note

====

> == Limitations

Reviewer: ```suggestion
[id="multi-tenancy-limitations"]
== Limitations
```

Achieving multi-tenancy within {product-title-short} is not possible for resources with a *global scope*.

> The resources with a global scope are:

Reviewer: ```suggestion
The following resources have a global scope:
```
